### ISSUE TITLE
Collision events are sent when not actually in contact

### ISSUE THREAD

**Issue by Jondolf** [Jondolf/avian#224]
Collision events are currently sent for contacts where the distance is below `NarrowPhaseConfig.prediction_distance`, but not necessarily touching. This can lead to unexpected event triggers as reported by `@COCPORN` in a situation where a ball sometimes bounces off of an object that despawns on contact and sometimes not.

Collision events should only be sent for contacts that are actually touching.

For now, you can get around the issue with this:

```rust
.insert_resource(NarrowPhaseConfig {
    prediction_distance: 0.0,
})
```

but it can cause contact stability issues in cases where moving bodies bump into each other.

**Comment by Technostalgic** [comment]
I also just want to say that it is actually quite nice to be able to query for "almost" collisions, like how it works now. Of course this should be fixed but there is also definitely a use case for the way that it works currently, as well. So maybe an unwarranted suggestion but maybe when this does get fixed, also add in a way to query for potential collisions within the prediction distance too?

Some use cases:

* Ground detection: It's nice to have a bit of leeway when performing ground detection, I personally just use contacts and coyote time, but I know raycasting is also a common method to look for ground detection, and this would be a similar way to do it as shape/ray casting, but better since you don't have to do an additional query, and it covers the exact shape of the colldier.  

* Wall Detection: Sometimes when directly next to a wall, if there isn't some force pressing you directly up against the wall, there will be no collision between the target body and the wall since the solver has moved the body to no longer be intersecting with the wall.

**Comment by Jondolf** [comment]
This should be fixed by #385. For those use cases I would still recommend using spatial queries like ray casts or shape casts, or even sensors that act as the "enlarged" collider.
